lib/web/ui/usercontext.go

```diff
@@ -84,9 +84,17 @@ func getAccessStrategy(roleset services.RoleSet) accessStrategy {
 			break
 		}
 
+		if strategy == types.RequestStrategyAlways {
+			continue
+		}
+
 		if options.RequestAccess == types.RequestStrategyAlways {
 			strategy = types.RequestStrategyAlways
 		}
+
+		if options.RequestAccess == types.RequestStrategyReasonRequired {
+			strategy = types.RequestStrategyReasonRequired
+		}
```

Comment:
```suggestion
		strategy = options.RequestAccess
```

do we need these conditionals? of the 3 remaining values it can be after the initial check in the loop, we just set it to whatever it is anyway (its already `Optional` when it starts)